File: src/org/recast4j/detour/extras/jumplink/Trajectory.java
package org.recast4j.detour.extras.jumplink;

public interface Trajectory {

    default float lerp(float f, float g, float u) {
        return u * g + (1f - u) * f;
    }

    float[] apply(Vector3f start, Vector3f end, float u);

}

File: src/org/recast4j/detour/extras/jumplink/Trajectory.kt
package org.recast4j.detour.extras.jumplink

import org.joml.Vector3f

interface Trajectory {
    companion object {
        fun lerp(a: Float, b: Float, t: Float) = a + (b - a) * t
    }

    fun apply(start: Vector3f, end: Vector3f, u: Float): Vector3f
}